test/state.c

```diff
@@ -36,6 +36,25 @@
  * keycode set (where ESC is 9). */
 #define EVDEV_OFFSET 8
 
+#define xkb_mod_index_t_true xkb_mod_index_t
```

Comment:
Heh it took me (more than...) a minute to understand what you did here :smile: 

I think it's a bit too clever. IIUC the main purpose of `make_mod_idx` is to add the assert, if so then I'd a static inline wrapper around `xkb_keymap_mod_get_index` which adds the assert, but I'd leave the declaration and assignment in the code, not in a macro.